raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -319,7 +319,19 @@ def handle_secretrequest(
 
     elif not is_valid_secretrequest and is_message_from_target:
         initiator_state.received_secret_request = True
-        iteration = TransitionResult(initiator_state, list())
+
+        if initiator_state.transfer_description.secret == EMPTY_SECRET:
+            transfer_description = initiator_state.transfer_description
+            payment_failed = EventPaymentSentFailed(
```

Comment:
I don't think this event should be used here. We can only guarantee that a payment has failed after the lock has expired and we expect this event to mean that.

The handling is probably better done on the `handle_secret_request` on the message handler, and do the same thing as the `handle_refund_transfer` (as in, handle the special case when the secret is unknown).

Reply:
@hackaugusto So you suggest that I will just ignore this request and let the transfer expire? 

Clearly I can do that but let's consider the app that made the payment (XUD). It can take some time until the transfer expires and in that time, XUD does not know what to do with the order in it order book. 
XUD knows for sure that it never released the secret  so there is no risk in getting back immediately and letting XUD knows that its peer is not configured right.

I added this handling  as it was hard to find what is causing XUD transfer to fail.   Keep in mind that this is a result of misconfiguration of the peer node and it is the sending that is suffering due to it. 

The handling we added in `handle_message_refundtransfer` is not similar. Here, if the first route fails the transfer is blocked until expiration.  However, on XUD level we are checking that there is a direct channel between the peers and that both peers are up, running, and connected to Raiden. So practically this situation of long payment can be avoided. 

Let me know what you think.

Reply:
@offerm I believe Augusto's argument was against using `EventPaymentSentFailed` as the event to indicate that the payment actually failed. At this point in time where you emit this event, the lockedd amounts are still locked and the payment never really failed in the sense that the locks no longer exist. At this point, we could end up in a situation where the secret request failed because the XUD did not provided it (False returned by the resolve)... so we send this event here but after a while the same request is sent and the resolver provides the secret and the payment was a success. Do you see where this is going? 
The alternative is as Augusto suggests, is to check this in the message handler to make sure that:
1. This is an atomic swap transfer (via secret being unknown to the node)
2. The resolver request fails
3. Iff both conditions above happen, then we never dispatch the secret request to the state machine.

At least, this is how i am thinking about it right now. Maybe Augusto has a better approach to how this can be done in details.

Reply:
As @rakanalh described, IMO the payment is failed only after the lock has expired, this is important because the event is tied to change in the channel capacity. Here is some relevant discussion we had in the past: https://github.com/raiden-network/raiden/pull/3024#discussion_r233371804 

If you want, you can add a new event as describe in the linked discussion. In your scenario since your application controls the secret, you can assume the payment will never succeed.